src/sysl/exporters/json_out/serializer.py

```diff
@@ -235,30 +315,116 @@ def deserializer(context):
     w()
     with java.Class(w, model_class + 'JsonDeserializer', write_file,
                     package=package,
-                    extends=has_tables and 'JsonDeserializer<' + model_class + '>'):
-        if has_tables:
+                    extends='JsonDeserializer<' + model_class + '>'):
+        w()
+        with java.Method(w, 'public', model_class, 'deserialize',
+                            [('JsonParser', 'p'),
+                             ('DeserializationContext', 'provider')],
+                            throws=['IOException', 'JsonParseException'], override=True):
+            w('{0} model = new {0}();', model_name)
+            if facade:
+                w('{0} facade = new {0}(model);', model_class)
             w()
-            with java.Method(w, 'public', model_class, 'deserialize',
-                             [('JsonParser', 'p'),
-                              ('DeserializationContext', 'provider')],
-                             throws=['IOException', 'JsonParseException'], override=True):
-                w('{0} model = new {0}();', model_name)
-                w()
-                w('eatToken(p, JsonToken.START_OBJECT);')
-                with java.While(w, 'p.getCurrentToken() != JsonToken.END_OBJECT'):
-                    with java.Switch(w, 'eatName(p)'):
-                        for (tname, t) in sorted(app.types.iteritems()):
-                            if t.HasField('relation'):
-                                w(('case "{0}": deserialize{0}Table(p, '
-                                   'model.get{0}Table()); break;'),
-                                  tname)
-                        w('default: skipJson(p);')
-                w('expectToken(p, JsonToken.END_OBJECT);')
-                w()
+            w('eatToken(p, JsonToken.START_OBJECT);')
+            with java.While(w, 'p.getCurrentToken() != JsonToken.END_OBJECT'):
+                with java.Switch(w, 'eatName(p)'):
+                    for (tname, t) in sorted(app.types.iteritems()):
+                        if t.HasField('relation'):
+                            w(('case "{0}": deserialize{0}Table(p, '
+                                'model.get{0}Table()); break;'),
+                                tname)
+                    w('default: skipJson(p);')
+            w('expectToken(p, JsonToken.END_OBJECT);')
+            w()
+            if facade:
+                w('return facade;')
+            else:
                 w('return model;')
 
         for (tname, t) in sorted(app.types.iteritems()):
             if not t.HasField('relation'):
+                if t.HasField('tuple'):
```

Comment:
Can you pull the last two lines together to avoid one indentation level as
`if not t.HasField('relation') and t.HasField('tuple'):` ?